mti_nma/bin/all.py

```diff
@@ -132,49 +151,83 @@ def run(
                         distributed_executor_address=distributed_executor_address,
                         clean=clean,
                         debug=debug,
                         struct=struct,
                         **kwargs
                     )
-                    avg_df = avgshape(
-                        sh_df=sh_df,
+                    avg_nuc_df = avgshape_nuc(
+                        sh_df=sh_nuc_df,
                         distributed_executor_address=distributed_executor_address,
                         clean=clean,
                         debug=debug,
                         struct=struct,
                         **kwargs
                     )
-                    nma(
-                        avg_df=avg_df,
+                    nma_nuc_df = nma_nuc(
+                        avg_df=avg_nuc_df,
                         distributed_executor_address=distributed_executor_address,
                         clean=clean,
                         debug=debug,
                         struct=struct,
                         **kwargs
                     )
 
-            # Run flow and get ending state
-            flow.run(executor=exe)
+                if "Cell" in structs:
+                    struct = "Cell"
 
-            if cell_flag:
-                draw_whist()
+                    sc_cell_df = single_cell(
+                        distributed_executor_address=distributed_executor_address,
+                        clean=clean,
+                        debug=debug,
+                        struct=struct,
+                        **kwargs
+                    )
+                    sh_cell_df = shparam_cell(
+                        sc_df=sc_cell_df,
+                        distributed_executor_address=distributed_executor_address,
+                        clean=clean,
+                        debug=debug,
+                        struct=struct,
+                        **kwargs
+                    )
+                    avg_cell_df = avgshape_cell(
+                        sh_df=sh_cell_df,
+                        distributed_executor_address=distributed_executor_address,
+                        clean=clean,
+                        debug=debug,
+                        struct=struct,
+                        **kwargs
+                    )
+                    nma_cell_df = nma_cell(
+                        avg_df=avg_cell_df,
+                        distributed_executor_address=distributed_executor_address,
+                        clean=clean,
+                        debug=debug,
+                        struct=struct,
+                        **kwargs
+                    )
+
+                # If nucleus and cell membrane were anlyzed, draw comparison plot
+                if "Nuc" and "Cell" in structs:
+                    compare_nuc_cell(nma_nuc_df, nma_cell_df)
+
+            # Run flow, get ending state, and visualize pipeline
+            flow.run(executor=exe)
+            flow.visualize()
```

Comment:
May want to remove this line or make it a parameter `--viz`. the dependencies required for `flow.visualize()` can't just be bare installed with `pip`.